iothub/device/src/Transport/Amqp/AmqpAuthenticationRefresher.cs

```diff
@@ -143,11 +174,20 @@ private async Task RefreshLoopAsync(DateTime refreshesOn, CancellationToken canc
         public void StopLoop()
         {
             if (Logging.IsEnabled)
+                Logging.Enter(this, nameof(StopLoop));
+
+            try
             {
-                Logging.Info(this, $"{nameof(StopLoop)}");
+                _refresherCancellationTokenSource?.Cancel();
+            }
```

Comment:
should it wait for the loop to end? We have cached the Task from it, right?

Reply:
We cannot wait for the loop to end since this is an infinite loop. Control will exit the loop only when the cancellation token is signaled for cancellation.

Reply:
This an unmonitored loop running on a separate thread.

Reply:
Right, but you just canceled it on line 181, right?

Reply:
I thought awaiting on the loop in here might help us catch an OperationCancelledException in the RefreshLoopAsync running unmonitored, but it turns out the only place we refer to the cancellation token is when we check it manually for cancellation. So, the API should not encounter OperationCancellationException.

Would there be any advantage to us awaiting this loop then?

Reply:
Awaiting for any exception and logging it.

If we don't await it, let's make sure that method does not throw, and logs anything it catches.

Reply:
Sure, updated accordingly.